library/kotlin/src/io/envoyproxy/envoymobile/EnvoyException.kt

```diff
@@ -0,0 +1,16 @@
+package io.envoyproxy.envoymobile
+
+import java.io.IOException
+
+/**
+ * An Envoy Exception class for describing what error may have occurred.
+ *
+ * @param code internal error code associated with the exception that occurred.
+ * @param message a description of what exception that occurred.
+ * @param cause an optional cause for the exception.
+ */
+class EnvoyException(
+    val code: Int,
```

Comment:
Is this an HTTP error code? If so should we make this a type that has bound checking as not all integers are valid HTTP error codes?

Reply:
This is to mirror what we have in the [`main_interface`](https://github.com/lyft/envoy-mobile/blob/master/library/common/main_interface.h#L84) and I tool out the `envoy` prefix since it's redundant.

Reply:
I'd suggest considering an `error_` prefix just to make it _very_ clear this is not an HTTP status code.

Reply:
I'm fine with changing this to `errorCode`, will update the Swift PR